frodo.js

```diff
@@ -1,58 +1,57 @@
-/**
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.